tests/number-test.toit

```diff
@@ -113,6 +117,16 @@ test-round:
   expect-number-out-of-range: float.INFINITY.round
   expect-number-out-of-range: float.NAN.round
 
+test-parse-number:
+  expect-identical 0 (num.parse "0")
+  expect-identical 0 (num.parse "-0")
+  expect-identical 0.0 (num.parse "0.0")
+
+  expect-identical int.MAX (num.parse "0b111111111111111111111111111111111111111111111111111111111111111")
+  expect-identical 9223372036854775808.0 (num.parse "9223372036854775808")
```

Comment:
Maybe add a comment that this is int.MAX + 1 as a float (if I am not mistaken)?

Reply:
done.